lib/src/license_detection/license.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:pana/src/license_detection/tokenizer.dart';
+
+@sealed
+class License {
+  /// Name of the license, is empty in case of unknown license.
+  final String identifier;
+
+  /// Original text from the license file.
+  final String content;
+
+  /// Normalized [Token]s created from the original text.
+  final List<Token> tokens;
+
+  /// A map of tokens and their count.
+  final Map<String, int> occurences;
+
+  License._(this.content, this.tokens, this.occurences, this.identifier);
+
+  factory License.parse(String identifier, String content) {
+    final tokens = tokenize(content);
+    final table = generateFrequencyTable(tokens);
+    return License._(content, tokens, table, identifier);
+  }
+}
+
+/// Contains deatils regarding the results of corpus license match with unknwown text.
+@sealed
+class LicenseMatch {
+  /// Sequence of tokens that were found in the unknown
+  ///  text that matched to tokens in any of the corpus license.
```

Comment:
Extra whitespace...
```suggestion
  /// Sequence of tokens that were found in the unknown
  /// text that matched to tokens in any of the corpus license.
```

Hmm, I'm not sure what to make of this:
> ... matched to tokens in any of the corpus license.

It might be better to say "matched license" rather than "corpus license".

Maybe, something like:
```suggestion
  /// Sequence of tokens from input text that is considered a match for the detected [license].
```